packages/snaps-controllers/src/snaps/endowments/name-lookup.ts

```diff
@@ -25,6 +26,48 @@ type NameLookupEndowmentSpecification = ValidPermissionSpecification<{
   validator: PermissionValidatorConstraint;
 }>;
 
+/**
+ * Helper function to determine if the passed in caveats
+ * match the specification for `endowment:name-lookup`.
+ *
+ * @param caveats - Value that needs to be validated.
+ * @returns boolean.
+ */
+
+const hasValidCaveats = (caveats: unknown) => {
```

Comment:
I've been trying to build a generic implementation of something like this in another PR: https://github.com/MetaMask/snaps/blob/5ae7801904c3985b6bfb03ec6a6d8a4d1b4d0c5e/packages/snaps-controllers/src/snaps/endowments/caveats/generic.ts#L21

Wondering if we should land that before this then 🤔

Reply:
I'd rather move this forward.

Reply:
I can understand that, but since we are gonna have more and more permissions with several caveats in the long-term I think it would be good to solve it more generically. 

Otherwise, we will land this PR and I will have to change this logic immediately after anyway 😓

Reply:
Unblocked now that I merged #2122